### ISSUE TITLE
Cannot install development version due to package mismatch

### ISSUE THREAD

**Issue by ghost** [shopware/platform#2037]
### PHP Version

7.4.22 (container's default version)

### Shopware Version

6.4.x-dev

### Expected behaviour

"./psh.phar install" executed within container should properly install and configure shopware 6 platform

### Actual behaviour

installator fails at this point
output is self explanatory

```
(36/52) Starting
> APP_ENV=prod bin/console framework:demodata
	
	Demodata Generator
	==================
	
	10:46:54 CRITICAL  [console] Error thrown while running command "framework:demodata". Message: "Class 'bheller\ImagesGenerator\ImagesGeneratorProvider' not found" ["exception" => Error { …},"command" => "framework:demodata","message" => "Class 'bheller\ImagesGenerator\ImagesGeneratorProvider' not found"]
	
	In DemodataService.php line 95:
	                                                                                               
	  Attempted to load class "ImagesGeneratorProvider" from namespace "bheller\ImagesGenerator".  
	  Did you forget a "use" statement for "Maltyxx\ImagesGenerator\ImagesGeneratorProvider"?      
	                                                                                               
	
	framework:demodata [-p|--products PRODUCTS] [-c|--categories CATEGORIES] [-o|--orders ORDERS] [-m|--manufacturers MANUFACTURERS] [-cs|--customers CUSTOMERS] [--media MEDIA] [--properties PROPERTIES] [--users USERS] [-ps|--product-streams PRODUCT-STREAMS] [-mt|--mail-template MAIL-TEMPLATE] [-mhf|--mail-header-footer MAIL-HEADER-FOOTER] [-y|--with-media [WITH-MEDIA]] [-r|--reviews [REVIEWS]] [--attribute-sets ATTRIBUTE-SETS] [--product-attributes PRODUCT-ATTRIBUTES] [--manufacturer-attributes MANUFACTURER-ATTRIBUTES] [--order-attributes ORDER-ATTRIBUTES] [--customer-attributes CUSTOMER-ATTRIBUTES] [--media-attributes MEDIA-ATTRIBUTES]
	
	
Execution aborted, a subcommand failed!
```

when I try to modify "maltyxx/images-generator": "1.0.0" in composer.json, then complete dependency hell starts

### How to reproduce

1. enter: https://github.com/shopware/platform
2. console: git@github.com:shopware/platform.git
3. console: ./psh.phar docker:start
4. console: ./psh.phar docker:ssh
5. console (within container): ./psh.phar install <- it fails

I'm running Debian Bullseye.

**Comment by shyim** [comment]
Run git checkout v6.4.3.1 in the dev template

**Comment by ghost** [comment]
Nothing changes

**Comment by shyim** [comment]
Remove composer.lock and vendor and run again psh install

**Comment by ghost** [comment]
nope, I got the same message

**Comment by ghost** [comment]
I got it somewhat working using this walkaround https://github.com/shopware/platform/issues/2003 however this issue is still not resolved.

**Comment by moorl** [comment]
trunk from yesterday and today same error in storefront (If Product Listing CMS Element bound):

```
An exception occurred while executing '# cms::product-listing::search-ids
SELECT `product`.`id` FROM `product` LEFT JOIN `product` `product.parent` ON `product`.`parent_id` = `product.parent`.`id` AND `product`.`version_id` = `product.parent`.`version_id` LEFT JOIN `product_visibility` `product.visibilities` ON `product`.`visibilities` = `product.visibilities`.`product_id` AND `product`.version_id = `product.visibilities`.product_version_id LEFT JOIN `product_category_tree` `product.categoriesRo.mapping` ON `product`.`id` = `product.categoriesRo.mapping`.`product_id` AND `product`.`version_id` = `product.categoriesRo.mapping`.`product_version_id` LEFT JOIN (SELECT `product.translation`.product_id, `product.translation`.product_version_id, `product.translation`.meta_description as `product.translation.metaDescription`, `product.translation`.name as `product.translation.name`, `product.translation`.keywords as `product.translation.keywords`, `product.translation`.description as `product.translation.description`, `product.translation`.meta_title as `product.translation.metaTitle`, `product.translation`.pack_unit as `product.translation.packUnit`, `product.translation`.pack_unit_plural as `product.translation.packUnitPlural`, `product.translation`.custom_search_keywords as `product.translation.customSearchKeywords`, `product.translation`.slot_config as `product.translation.slotConfig`, `product.translation`.custom_fields as `product.translation.customFields`, `product.translation`.created_at as `product.translation.createdAt`, `product.translation`.updated_at as `product.translation.updatedAt`, `product.translation`.product_id as `product.translation.productId`, `product.translation`.language_id as `product.translation.languageId`, `product.translation`.product_version_id as `product.translation.productVersionId` FROM `product_translation` `product.translation` WHERE `product.translation`.language_id = ?) `product.product_translation` ON `product.product_translation`.`product_id` = `product`.`id` AND `product.product_translation`.product_version_id = `product`.version_id LEFT JOIN (SELECT `product.parent.translation`.product_id, `product.parent.translation`.product_version_id, `product.parent.translation`.meta_description as `product.parent.translation.metaDescription`, `product.parent.translation`.name as `product.parent.translation.name`, `product.parent.translation`.keywords as `product.parent.translation.keywords`, `product.parent.translation`.description as `product.parent.translation.description`, `product.parent.translation`.meta_title as `product.parent.translation.metaTitle`, `product.parent.translation`.pack_unit as `product.parent.translation.packUnit`, `product.parent.translation`.pack_unit_plural as `product.parent.translation.packUnitPlural`, `product.parent.translation`.custom_search_keywords as `product.parent.translation.customSearchKeywords`, `product.parent.translation`.slot_config as `product.parent.translation.slotConfig`, `product.parent.translation`.custom_fields as `product.parent.translation.customFields`, `product.parent.translation`.created_at as `product.parent.translation.createdAt`, `product.parent.translation`.updated_at as `product.parent.translation.updatedAt`, `product.parent.translation`.product_id as `product.parent.translation.productId`, `product.parent.translation`.language_id as `product.parent.translation.languageId`, `product.parent.translation`.product_version_id as `product.parent.translation.productVersionId` FROM `product_translation` `product.parent.translation` WHERE `product.parent.translation`.language_id = ?) `product.product_translation.parent` ON `product.product_translation.parent`.`product_id` = `product.parent`.`id` AND `product.product_translation.parent`.product_version_id = `product.parent`.version_id LEFT JOIN `category` `product.categoriesRo` ON `product.categoriesRo.mapping`.`category_id` = `product.categoriesRo`.`id` AND `product.categoriesRo`.`version_id` = `product.categoriesRo.mapping`.`category_version_id` WHERE (`product`.`version_id` = ?) AND ((((`product.visibilities`.`visibility` >= ?) AND `product.visibilities`.`sales_channel_id` = ? AND IFNULL(`product`.`active`, `product.parent`.`active`) = ?) AND `product.categoriesRo`.`id` = ? AND NOT (`product`.`display_group` <=> ?))) GROUP BY `product`.`display_group` ORDER BY MIN(COALESCE(`product.translation.name`,`product.parent.translation.name`)) ASC LIMIT 24' with params ["\x2f\xbb\x5f\xe2\xe2\x9a\x4d\x70\xaa\x58\x54\xce\x7c\xe3\xe2\x0b", "\x2f\xbb\x5f\xe2\xe2\x9a\x4d\x70\xaa\x58\x54\xce\x7c\xe3\xe2\x0b", "\x0f\xa9\x1c\xe3\xe9\x6a\x4b\xc2\xbe\x4b\xd9\xce\x75\x2c\x34\x25", 30, "\x59\xc6\x7c\xdc\xa0\xbc\x45\x4d\xb1\x79\x9a\x0f\x1a\xcd\x9f\x8e", true, "\xb2\x14\xaf\xb8\x30\x2c\x44\xa9\x92\x2e\x01\x29\x15\x55\xe1\x21", null]:

SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'shopware.product.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```

**Comment by shyim** [comment]
@moorl Can you revert this? https://github.com/shopware/platform/commit/f6029ecb0087e3ec829832dcc555071b3427ada0

**Comment by moorl** [comment]
@shyim 
Thanks, removing this Query from Kernel.php made it work for me.

![image](https://user-images.githubusercontent.com/45460948/130848187-c896308a-3f99-48e8-b297-91169390c99d.png)

**Comment by shyim** [comment]
I hate my life. will revert it tomorrow

**Comment by moorl** [comment]
https://github.com/shopware/platform/issues/1842

Just an Idea:

Make a Shop Configuration for all given Timezones known in PHP

```XML
        <input-field type="text">
            <name>timezone</name>
            <label>Timezone of your Shop</label>
            <label lang="de-DE">Zeitzone des Shops</label>
            <helpText>Set your time zone for the store here.</helpText>
            <helpText lang="de-DE">Setzen Sie hier Ihre Zeitzone für den Shop.</helpText>
            <defaultValue>Europe/Berlin</defaultValue>
            <placeholder>Europe/Berlin</placeholder>
        </input-field>
```

And place the Config Variable here...
https://github.com/shopware/platform/blob/84d0afbd23356d7b4ea20b2cabbe76ba742f5f0e/src/Core/Framework/DataAbstractionLayer/FieldSerializer/DateTimeFieldSerializer.php#L43

Could it work?

**Comment by shyim** [comment]
Please create a new issue for that. that doesn't fit here :D

**Comment by moorl** [comment]
Alright, thank you anyway :)

**Comment by xUJYx** [comment]
@R0b0c0p @shyim 
The error occurs cause development part of shopware accepted [this pull-request](https://github.com/shopware/development/pull/159). They have changed composer dependencies there, but those dependencies are required here at platform part...

So if you want to start work with development version - you need to make these commands after installing development shopware:
`composer remove maltyxx/images-generator fakerphp/faker mbezhanov/faker-provider-collection`
`composer require bheller/images-generator:1.0.1 fzaninotto/faker:1.9.1 mbezhanov/faker-provider-collection:1.2.1 --dev`

And after that start your docker commands
`console: ./psh.phar docker:start`
`console: ./psh.phar docker:ssh`
`console (within container): ./psh.phar install <- AND it will now NOT fails`

Or just wait when contributors of platform part of shopware will update this dependencies...

**Comment by shyim** [comment]
Hey @xUJYx ,

It is intended how it should be. The `trunk` branch is developed against `trunk` of the platform. When you need an older version checkout a tag like `6.4.3.1`. See composer.json of that version: https://github.com/shopware/platform/blob/v6.4.3.1/composer.json

**Comment by xUJYx** [comment]
> When you need an older version checkout a tag like `6.4.3.1`. See composer.json of that version: https://github.com/shopware/platform/blob/v6.4.3.1/composer.json

Hi @shyim 
There's no need in old version. There's a need of 'working scripts' when doing step-by-step all instruction from docs.

Current situation: if you'll try to start development version of shopware step by step - you'll fail...

Trunk branch of development has conflict dependency with trunk branch of platform...

**Comment by shyim** [comment]
The dev template takes by default the newest released version which is 6.4.3.1 currently. To have a development repo trunk with a platform repo trunk you need to delete the platform folder in the development repository and clone it there and run composer update.

Monday will be 6.4.4.0 released, where the trunk of the development branch will work again with the latest released version

**Comment by xUJYx** [comment]
> Monday will be 6.4.4.0 released, where the trunk of the development branch will work again with the latest released version

@shyim great to hear that. 
Thanks

**Comment by shyim** [comment]
Should be solved after the release of 6.4.4.0
